src/passes/hash-stringify-walker.cpp

```diff
@@ -68,4 +68,33 @@ void HashStringifyWalker::visitExpression(Expression* curr) {
   }
 }
 
+std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::dedupe(
+  const std::vector<SuffixTree::RepeatedSubstring> substrings) {
+  std::set<uint32_t> seen;
+  std::vector<SuffixTree::RepeatedSubstring> result;
+  for (auto substring : substrings) {
+    std::vector<uint32_t> idxToInsert;
+    bool seenEndIdx = false;
+    for (auto startIdx : substring.StartIndices) {
+      // We are using the end index to ensure that each repeated substring
+      // reported by the SuffixTree is unique. This is because LLVM's SuffixTree
+      // reports back repeat sequences that are substrings of longer repeat
+      // sequences with the same endIdx, and we generally prefer to outline
+      // longer repeat sequences.
+      uint32_t endIdx = substring.Length + startIdx;
+      if (auto found = seen.find(endIdx); found != seen.end()) {
+        seenEndIdx = true;
+        continue;
```

Comment:
It looks like this can be a `break` since `idxToInsert` is not used if we arrive here.

Reply:
done